### PR TITLE
feat: P2022-364 sign up form

### PR DESCRIPTION
**Pull request by lenasokolova**

### CODE REVIEW

File: app/components/Form/index.js [hunk 1]
```diff
@@ -15,8 +15,97 @@ export function Input(props) {
...
+    '^[0-9a-zA-Z]{6,20}$'
+);
+
+export const validPassword = new RegExp(
+    '^[0-9a-zA-Z]{6,20}$'
+);
+
+export const validEmail =  new RegExp(
+    "^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:.[a-zA-Z0-9-]+)*$" // Użyty stary RegExp - niepoprawny
+);
```
Comment (hiimsagan) [inline app/components/Form/index.js:110]: Soo is it working right now ?

File: styles/formStyles.module.css [hunk 2]
```diff
@@ -0,0 +1,6 @@
+.inputStyle {
+    background-color: #f2f2f2;
+    border: none;
+    height: 25px;
+    margin-bottom: 5px;
+}
```
Comment (hiimsagan) [inline styles/formStyles.module.css:6]: Is this necessary when you don't used it ?

File: pages/user/register.js [hunk 3]
```diff
@@ -1,7 +1,98 @@
...
+        if (!validName.test(username))
+        {
+            setUsernameError('Username must be between 6 and 20 characters, letters and numbers are only allowed');
+        }
+
+        setPasswordError('');
+        if (!validPassword.test(password))
+        {
+            setPasswordError('Password must be at least 6 characters long')
+        }
```
Comment (rast-mk1) [inline pages/user/register.js:32]: User is not informed, that he can use only [0-9a-zA-Z] chars and the password must be {6,20} chars long

File: pages/user/register.js [hunk 4]
```diff
@@ -1,7 +1,98 @@
...
+            setEmailError('Invalid email format')
+        }
+
+        setConfEmailError('');
+        if (email !== confEmail)
+        {
+            setConfEmailError("Emails don't match")
+        } else{
+            setMessage(alert(message))
+        }
```
Comment (MS151994) [inline pages/user/register.js:46]: Why are you checking the email and not adding it to the object?

File: pages/user/register.js [hunk 5]
```diff
@@ -1,7 +1,98 @@
...
+        {
+            setEmailError('Invalid email format')
+        }
+
+        setConfEmailError('');
+        if (email !== confEmail)
+        {
+            setConfEmailError("Emails don't match")
+        } else{
+            setMessage(alert(message))
```
Comment (MS151994) [inline pages/user/register.js:45]: What is the "message"? and why it is "undefined"?

File: pages/user/register.js (hunk 4 above)
Comment (rast-mk1) [inline pages/user/register.js:46]: not sure if setMessage should have alert() in it; 

also, only email and confEmail are taken into consideration, but username validation is skipped

File: app/components/Form/index.js [hunk 6]
```diff
@@ -15,8 +15,97 @@ export function Input(props) {
...
+        
+        </>
+    )
+}
+
+export const InputStyle = styled.input`
+    background-color: #f2f2f2;
+    border: none;
+    height: 25px;
+`;
```
Comment (marcelina-hasiak) [inline app/components/Form/index.js:41]: If you'd like to use this component inside `InputReg`, just switch your `<input>` tag with `<InputStyle>`

File: app/components/Form/index.js [hunk 7]
```diff
@@ -15,8 +15,97 @@ export function Input(props) {
...
+    &:focus{
+        background-color: #fff;
+        color: green;
+        border: 2px solid green;
+
+    }
+`;
+
+
+// RegEx component
```
Comment (weronika299) [inline app/components/Form/index.js:98]: You have a lot of empty lines in your file. Maybe it's better to remove them for better code readability.

**Pull request closed without merging**